### PR TITLE
[GPU] Refactor

### PR DESCRIPTION
**Pull request by andrei-cv**
### Details:
 - *activation*

### Tickets:
 - *CVS-124001*

### CODE REVIEW

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/activation.cpp [hunk 1]
```diff
@@ -2,131 +2,121 @@
...
+using ov::test::ActivationLayerTest;
+using ov::test::ActivationParamLayerTest;
+using ov::test::utils::ActivationTypes;
 
-const std::vector<InferenceEngine::Precision> netPrecisions = {
-        InferenceEngine::Precision::FP32,
-        InferenceEngine::Precision::FP16
+// Common params
+const std::vector<ov::element::Type> netPrecisions = {
+        //ov::element::f32,
```
Comment (vurusovs) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/activation.cpp:15]: Could you comment why f32 is removed?

File: src/tests/ov_helpers/ov_models/src/activation.cpp [hunk 2]
```diff
@@ -63,7 +63,7 @@ std::shared_ptr<ov::Node> makeActivation(const ov::Output<Node>& in,
     case ov::test::utils::ActivationTypes::Relu:
         return std::make_shared<ov::op::v0::Relu>(in);
     case ov::test::utils::ActivationTypes::LeakyRelu: {
-        auto leaky_slope = std::make_shared<ov::op::v0::Constant>(ov::element::f32, inShape, constantsValue);
```
Comment (vurusovs) [inline src/tests/ov_helpers/ov_models/src/activation.cpp:66]: @itikhono @iefode please, review this file

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp [hunk 3]
```diff
@@ -190,7 +190,7 @@ void inline fill_data_random(T* pointer,
         start_from = 0;
     }
     for (std::size_t i = 0; i < size; i++) {
-        pointer[i] = static_cast<T>(start_from + static_cast<T>(random.Generate(k_range)) / k);
```
Comment (vurusovs) [inline src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:193]: @andrei-cv is it required? Let's revert

Comment (andrei-cv) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:193]: random.Generate overflows fp16

Comment (vurusovs) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:193]: Got it, LGTM
@iefode please, take a look

**Review by Lyamin-Roman — APPROVED** [review]
Plugin part LGTM

**Review by vurusovs — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp (hunk 3 above)
Comment (iefode) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:193]: @andrei-cv @vurusovs 
Opened questions:
* What if we support FP64?
* moreover float (f32) > fp16...
* we cast values to `<T>`

File: src/tests/ov_helpers/ov_models/src/activation.cpp (hunk 2 above)
Comment (iefode) [reply src/tests/ov_helpers/ov_models/src/activation.cpp:66]: LGTM

**Review by iefode — APPROVED** [review]

File: src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp (hunk 3 above)
Comment (vurusovs) [reply src/tests/test_utils/common_test_utils/include/common_test_utils/data_utils.hpp:193]: Ticket for AR: [CVS-125993](https://jira.devtools.intel.com/browse/CVS-125993)

**Pull request closed without merging**